### PR TITLE
bump Cluster API to v1.1.0

### PR DESCRIPTION
**Pull request by ykakarap**
### What this PR does / why we need it

This PR update the version of ClusterAPI to v1.1.0.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by stmcginnis** [comment]
> Not sure if this will be an issue for us, but we may want to wait until resolution of this issue: https://github.com/kubernetes-sigs/cluster-api/issues/6093

CAPI 1.1.1 has now been released to address that problem: https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.1.1

File: pkg/v1/providers/cluster-api/v1.1.0/core-components.yaml [hunk 1]
```diff
@@ -0,0 +1,10124 @@
...
+          status:
+            description: MachineDeploymentStatus defines the observed state of MachineDeployment.
+            properties:
+              availableReplicas:
+                description: Total number of available machines (ready for at least
+                  minReadySeconds) targeted by this deployment.
+                format: int32
+                type: integer
+              conditions:
+                descri
```
Comment (anujc25) [inline pkg/v1/providers/cluster-api/v1.1.0/core-components.yaml]: We need to enable `CLUSTER_TOPOLOGY` by default.

**Review by anujc25 — COMMENTED** [review]
I think we need to enable the `CLUSTER_TOPOLOGY` feature by default. Currently it is disabled by default.

```
- --feature-gates=MachinePool=${EXP_MACHINE_POOL:=false},ClusterResourceSet=${EXP_CLUSTER_RESOURCE_SET:=true},ClusterTopology=${CLUSTER_TOPOLOGY:=false}

```

**Comment by ykakarap** [comment]
@stmcginnis Since v1.1.1 is released I am going to directly bump to v1.1.1.

**Comment by ykakarap** [comment]
Closing this in favor of https://github.com/vmware-tanzu/tanzu-framework/pull/1669.

We are going to directly upgrade to Cluster API v1.1.1 that includes an important KCP fix (ref: #https://github.com/kubernetes-sigs/cluster-api/issues/6093)

**Pull request closed without merging**